### ISSUE TITLE
[Bug] MasterDetailPage (and FlyoutPage) cannot customize status bar text color on iOS

### ISSUE THREAD

**Issue by cabal95** [xamarin/Xamarin.Forms#13053]
### Description

TL;DR: There doesn't seem to be any way to set the status bar text to use LightContent (white text) when using MasterDetailPage or the new FlyoutPage on iPad.

This MAY be a new issue since iOS 14, I'm not 100% sure. I don't have a pre-iOS 14 to test on at the moment. But it looks like iOS 14 added special internal code when dealing with UISplitViewController to allow each child view controller to specify it's own status bar text tinting (light or dark). Otherwise this has just always been a bug and nobody has run into it before.

So it appears that, and again this may be new in iOS 14, on iOS when using a UISplitViewController the PreferredStatusBarStyle() method is called, but it's value is ignored. Instead, the OS seems to then call the CHILD view controllers' PreferredStatusBarStyle() methods. Because a MasterDetailPage and FlyoutPage renderers use a custom internal ChildViewController and EventedViewController class as their direct children - which then contain the user-supplied view controller renderers - those calls are swallowed and the default value is returned.

The benefit of the way Apple is doing this is that each child ViewController can specific it's own status bar text color. So for example, with the Xamarin.Forms.ControlGallery.iOS sample app, it uses a split view with two different background colors. You can now specify that the left one should be white text on the cyan background and the right view should be black text on the white background.

But since our methods aren't ever called, we can't actually do that. The solution seems to be to add this method to the ChildViewController internal class:

```cs
public override UIViewController ChildViewControllerForStatusBarStyle()
{
	return ChildViewControllers?.LastOrDefault();
}
```

Though, I think a number of those "ChildViewControllerFor..." methods that are implemented in PlatformRenderer should probably be replicated into this class. But, with the above change I was instantly able to control the status bar text color in my application. Unfortunately that isn't a change I can make myself without rolling my own custom build of Xamarin Forms, which I don't really want to do.

### Steps to Reproduce

1. Create a forms app that uses a MasterDetailPage
2. Add child pages that each have a dark blue background color
3. Attempt to change the status bar text color by overriding PreferredStatusBarStyle in custom renderers.
4. Fail.

### Expected Behavior

Those method calls should be passed down to the user-supplied child page renderers so they can be implemented.

### Actual Behavior

They are not, causing those queries from iOS to be silently swallowed.

### Basic Information

- Version with issue: 4.6.0 - 5.0.0
- Last known good version: Always been this way
- Platform Target Frameworks:
  - iOS:  Latest - probably 14.2?
  - Android: n/a
  - UWP:  n/a
- Android Support Library / AndroidX Version: n/a
- NuGet Packages: n/a
- Affected Devices: Apple iPad

### Environment

```
Microsoft Visual Studio Professional 2019
Version 16.8.2
VisualStudio.16.Release/16.8.2+30717.126
Microsoft .NET Framework
Version 4.8.03752

Installed Version: Professional

Visual C++ 2019   00435-30012-51493-AA611
Microsoft Visual C++ 2019

ASP.NET and Web Tools 2019   16.8.553.28003
ASP.NET and Web Tools 2019

ASP.NET Core Razor Language Services   16.1.0.2052803+84e121f1403378489b842e1797df2f3f5a49ac3c
Provides languages services for ASP.NET Core Razor.

ASP.NET Web Frameworks and Tools 2019   16.8.553.28003
For additional information, visit https://www.asp.net/

Azure App Service Tools v3.0.0   16.8.553.28003
Azure App Service Tools v3.0.0

Azure Functions and Web Jobs Tools   16.8.553.28003
Azure Functions and Web Jobs Tools

C# Tools   3.8.0-5.20567.16+53c5d7d3cf13d88978744a32a27c5f8350a8400a
C# components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Common Azure Tools   1.10
Provides common services for use by Azure Mobile Services and Microsoft Azure Tools.

Extensibility Message Bus   1.2.6 (master@34d6af2)
Provides common messaging-based MEF services for loosely coupled Visual Studio extension components communication and integration.

Generate GUID for Visual Studio 2012   1.0
Generate a variety of GUID values with a click or by assigning a chord to the menu command with the ability to change formats using a drop down in the Tools menu.

GhostDoc   20.2.20270.0
Generate XML Comments from your code, maintain clean and up-to-date documentation, produce help documentation in multiple formats, use intelligent source code Spell Checker in Visual Studio.

IntelliCode Extension   1.0
IntelliCode Visual Studio Extension Detailed Info

Microsoft Azure Tools   2.9
Microsoft Azure Tools for Microsoft Visual Studio 2019 - v2.9.30924.1

Microsoft Continuous Delivery Tools for Visual Studio   0.4
Simplifying the configuration of Azure DevOps pipelines from within the Visual Studio IDE.

Microsoft JVM Debugger   1.0
Provides support for connecting the Visual Studio debugger to JDWP compatible Java Virtual Machines

Microsoft Library Manager   2.1.113+g422d40002e.RR
Install client-side libraries easily to any web project

Microsoft MI-Based Debugger   1.0
Provides support for connecting Visual Studio to MI compatible debuggers

Microsoft Visual C++ Wizards   1.0
Microsoft Visual C++ Wizards

Microsoft Visual Studio Tools for Containers   1.1
Develop, run, validate your ASP.NET Core applications in the target environment. F5 your application directly into a container with debugging, or CTRL + F5 to edit & refresh your app without having to rebuild the container.

Microsoft Visual Studio VC Package   1.0
Microsoft Visual Studio VC Package

Mono Debugging for Visual Studio   16.8.43 (00471f8)
Support for debugging Mono processes with Visual Studio.

NuGet Package Manager   5.8.0
NuGet Package Manager in Visual Studio. For more information about NuGet, visit https://docs.nuget.org/

ProjectServicesPackage Extension   1.0
ProjectServicesPackage Visual Studio Extension Detailed Info

SQL Server Data Tools   16.0.62010.06180
Microsoft SQL Server Data Tools

TypeScript Tools   16.0.21016.2001
TypeScript Tools for Microsoft Visual Studio

Visual Basic Tools   3.8.0-5.20567.16+53c5d7d3cf13d88978744a32a27c5f8350a8400a
Visual Basic components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Visual F# Tools   16.8.0-beta.20507.4+da6be68280c89131cdba2045525b80890401defd
Microsoft Visual F# Tools

Visual Studio Code Debug Adapter Host Package   1.0
Interop layer for hosting Visual Studio Code debug adapters in Visual Studio

Visual Studio Container Tools Extensions   1.0
View, manage, and diagnose containers within Visual Studio.

Visual Studio Spell Check Everywhere   VSSpellCheckEverywhere
An extension that enables spell checking within any Visual Studio file editor or tool window that uses WPF text boxes.
https://GitHub.com/EWSoftware/VSSpellChecker

Visual Studio Spell Checker   VSSpellChecker
An editor extension that checks the spelling of comments, strings, and plain text as you type or interactively with tool windows.
https://GitHub.com/EWSoftware/VSSpellChecker

Visual Studio Tools for Containers   1.0
Visual Studio Tools for Containers

VisualStudio.DeviceLog   1.0
Information about my package

VisualStudio.Foo   1.0
Information about my package

VisualStudio.Mac   1.0
Mac Extension for Visual Studio

Xamarin   16.8.000.255 (d16-8@d002176)
Visual Studio extension to enable development for Xamarin.iOS and Xamarin.Android.

Xamarin Designer   16.8.0.507 (remotes/origin/d16-8@e87b24884)
Visual Studio extension to enable Xamarin Designer tools in Visual Studio.

Xamarin Templates   16.8.112 (86385a3)
Templates for building iOS, Android, and Windows apps with Xamarin and Xamarin.Forms.

Xamarin.Android SDK   11.1.0.17 (d16-8/c0e2b8e)
Xamarin.Android Reference Assemblies and MSBuild support.
    Mono: be2226b
    Java.Interop: xamarin/java.interop/d16-8@79d9533
    ProGuard: Guardsquare/proguard/proguard6.2.2@ebe9000
    SQLite: xamarin/sqlite/3.32.1@1a3276b
    Xamarin.Android Tools: xamarin/xamarin-android-tools/d16-8@2fb1cbc


Xamarin.iOS and Xamarin.Mac SDK   14.4.1.3 (e30c41de3)
Xamarin.iOS and Xamarin.Mac Reference Assemblies and MSBuild support.
```

### Screenshots

![2020-12-04_11-17-19-AM](https://user-images.githubusercontent.com/1119863/101199608-623ec180-3622-11eb-9a87-82273346ae40.png)

### Reproduction Link

[App2.zip](https://github.com/xamarin/Xamarin.Forms/files/5644596/App2.zip)

App includes an example of failing to be able to set status bar text color when using MasterDetailPage on iPad. It also includes showing a working example (if uncommented) when just using a plain ContentPage on iPad.

### Workaround

None currently found.

**Comment by knasher** [comment]
I left a comment on the PR to fix this bug but in case it is missed I wanted to ask it here too, would a fix also be needed when using the Shell? I have a Shell based app and cannot control the status bar test color with PreferredStatusBarStyle on any page that is loaded. I've tried a number of renderers for different elements and at no point will the method be called, is this a potential Xamarin bug?

**Comment by knasher** [comment]
For anyone who might be having the same problem with Shell, I have created a bug #14246 and a PR #14247
